File: data/imported/knowledge/cells/typescript/merged/snippets/snippet_023.ts
async function testWorksheetCopy() {
  const workbook = new Workbook();

  const ws1 = workbook.worksheets[0]!;
  ws1.putValue("A1", "Original Data");
  ws1.putValue("A2", "Row 2");

  const ws2 = workbook.worksheets.addWorksheet("CopySheet");
  ws2.putValue("A1", "Copied Data");

  console.log(
    "Worksheets:",
    workbook.worksheets.worksheets.map((w) => w.name),
  );

  await workbook.save("outputfiles/test_worksheet_copy.xlsx");
  console.log("Saved to outputfiles/test_worksheet_copy.xlsx");
}

File: data/imported/knowledge/cells/python/merged/snippets/snippet_023.py
# Adapted from aspose.org: knowledge/cells/python/merged/snippets/snippet_023.py @ 7f72da4e1423546104b40fa8cebf5b9ae3ce9c91
# Imported under the authorization recorded in
# plans/investigations/evidence/imported-corpus-v1/licensing-resolution-state.md

def test_remove_auto_filter_persistence(self):

        """Test that removing auto filter is persisted correctly."""

        wb = Workbook()

        ws = wb.worksheets[0]

        

        # Add some data

        ws.cells['A1'].value = "Name"

        ws.cells['B1'].value = "Age"

        ws.cells['A2'].value = "Alice"

        ws.cells['B2'].value = 30

        

        # Set auto filter range and apply filter

        ws.auto_filter.range = "A1:B2"

        ws.auto_filter.filter(0, ["Alice"])

        

        # Remove auto filter

        ws.auto_filter.remove()

        

        # Save and reload

        test_file = os.path.join(self.output_dir, 'test_remove_auto_filter.xlsx')

        wb.save(test_file)

        wb2 = Workbook(test_file)

        ws2 = wb2.worksheets[0]

        

        # Verify auto filter is removed

        self.assertIsNone(ws2.auto_filter.range)

        self.assertEqual(ws2.auto_filter.filter_columns, {})